File: dmn-test-cases/signavio/dmn/dmn2java/expected/proto/proto3/java/dmn/example-credit-decision/type/GenerateOutputData.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinitionInterface.ftl", "generateOutputData"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(as = type.GenerateOutputDataImpl.class)
public interface GenerateOutputData extends com.gs.dmn.runtime.DMNType {
    static GenerateOutputData toGenerateOutputData(Object other) {
        if (other == null) {
            return null;
        } else if (GenerateOutputData.class.isAssignableFrom(other.getClass())) {
            return (GenerateOutputData)other;
        } else if (other instanceof com.gs.dmn.runtime.Context) {
            GenerateOutputDataImpl result_ = new GenerateOutputDataImpl();
            if (((com.gs.dmn.runtime.Context)other).keySet().contains("decision") || ((com.gs.dmn.runtime.Context)other).keySet().contains("Decision")) {
                result_.setDecision((String)((com.gs.dmn.runtime.Context)other).get("decision", "Decision"));
            } else {
                return  null;
            }
            if (((com.gs.dmn.runtime.Context)other).keySet().contains("assessment") || ((com.gs.dmn.runtime.Context)other).keySet().contains("Assessment")) {
                result_.setAssessment((java.lang.Number)((com.gs.dmn.runtime.Context)other).get("assessment", "Assessment"));
            } else {
                return  null;
            }
            if (((com.gs.dmn.runtime.Context)other).keySet().contains("issue") || ((com.gs.dmn.runtime.Context)other).keySet().contains("Issue")) {
                result_.setIssue((java.lang.Number)((com.gs.dmn.runtime.Context)other).get("issue", "Issue"));
            } else {
                return  null;
            }
            return result_;
        } else if (other instanceof com.gs.dmn.runtime.DMNType) {
            return toGenerateOutputData(((com.gs.dmn.runtime.DMNType)other).toContext());
        } else if (other instanceof proto.GenerateOutputData) {
            GenerateOutputDataImpl result_ = new GenerateOutputDataImpl();
            result_.setDecision(((proto.GenerateOutputData) other).getDecision());
            result_.setAssessment(((java.lang.Number) java.math.BigDecimal.valueOf(((proto.GenerateOutputData) other).getAssessment())));
            result_.setIssue(((java.lang.Number) java.math.BigDecimal.valueOf(((proto.GenerateOutputData) other).getIssue())));
            return result_;
        } else {
            throw new com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.getClass().getSimpleName(), GenerateOutputData.class.getSimpleName()));
        }
    }

    static proto.GenerateOutputData toProto(GenerateOutputData other) {
        proto.GenerateOutputData.Builder result_ = proto.GenerateOutputData.newBuilder();
        if (other != null) {
            String decisionProto_ = (((GenerateOutputData) other).getDecision() == null ? "" : ((GenerateOutputData) other).getDecision());
            result_.setDecision(decisionProto_);
            Double assessmentProto_ = (((GenerateOutputData) other).getAssessment() == null ? 0.0 : ((GenerateOutputData) other).getAssessment().doubleValue());
            result_.setAssessment(assessmentProto_);
            Double issueProto_ = (((GenerateOutputData) other).getIssue() == null ? 0.0 : ((GenerateOutputData) other).getIssue().doubleValue());
            result_.setIssue(issueProto_);
        }
        return result_.build();
    }

    static List<proto.GenerateOutputData> toProto(List<GenerateOutputData> other) {
        if (other == null) {
            return null;
        } else {
            return other.stream().map(o -> toProto(o)).collect(java.util.stream.Collectors.toList());
        }
    }

    @com.fasterxml.jackson.annotation.JsonGetter("Decision")
    String getDecision();

    @com.fasterxml.jackson.annotation.JsonGetter("Assessment")
    java.lang.Number getAssessment();

    @com.fasterxml.jackson.annotation.JsonGetter("Issue")
    java.lang.Number getIssue();

    default com.gs.dmn.runtime.Context toContext() {
        com.gs.dmn.runtime.Context context = new com.gs.dmn.runtime.Context();
        context.put("decision", getDecision());
        context.put("assessment", getAssessment());
        context.put("issue", getIssue());
        return context;
    }

    default boolean equalTo(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        GenerateOutputData other = (GenerateOutputData) o;
        if (this.getAssessment() != null ? !this.getAssessment().equals(other.getAssessment()) : other.getAssessment() != null) return false;
        if (this.getDecision() != null ? !this.getDecision().equals(other.getDecision()) : other.getDecision() != null) return false;
        if (this.getIssue() != null ? !this.getIssue().equals(other.getIssue()) : other.getIssue() != null) return false;

        return true;
    }

    default int hash() {
        int result = 0;
        result = 31 * result + (this.getAssessment() != null ? this.getAssessment().hashCode() : 0);
        result = 31 * result + (this.getDecision() != null ? this.getDecision().hashCode() : 0);
        result = 31 * result + (this.getIssue() != null ? this.getIssue().hashCode() : 0);
        return result;
    }

    default String asString() {
        StringBuilder result_ = new StringBuilder("{");
        result_.append("Assessment=" + getAssessment());
        result_.append(", Decision=" + getDecision());
        result_.append(", Issue=" + getIssue());
        result_.append("}");
        return result_.toString();
    }
}

File: dmn-signavio/src/test/resources/dmn2java/exported/proto/expected/proto3/kotlin/dmn/example-credit-decision/type/GenerateOutputData.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinitionInterface.ftl", "generateOutputData"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(`as` = type.GenerateOutputDataImpl::class)
interface GenerateOutputData : com.gs.dmn.runtime.DMNType {
    @get:com.fasterxml.jackson.annotation.JsonGetter("Decision")
    val decision: String?

    @get:com.fasterxml.jackson.annotation.JsonGetter("Assessment")
    val assessment: java.math.BigDecimal?

    @get:com.fasterxml.jackson.annotation.JsonGetter("Issue")
    val issue: java.math.BigDecimal?

    override fun toContext(): com.gs.dmn.runtime.Context {
        val context = com.gs.dmn.runtime.Context()
        context.put("decision", this.decision)
        context.put("assessment", this.assessment)
        context.put("issue", this.issue)
        return context
    }

    fun equalTo(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as GenerateOutputData
        if (if (this.assessment != null) this.assessment != other.assessment else other.assessment != null) return false
        if (if (this.decision != null) this.decision != other.decision else other.decision != null) return false
        if (if (this.issue != null) this.issue != other.issue else other.issue != null) return false

        return true
    }

    fun hash(): Int {
        var result = 0
        result = 31 * result + (if (this.assessment != null) this.assessment.hashCode() else 0)
        result = 31 * result + (if (this.decision != null) this.decision.hashCode() else 0)
        result = 31 * result + (if (this.issue != null) this.issue.hashCode() else 0)
        return result
    }

    fun asString(): String {
        val result_ = StringBuilder("{")
        result_.append("Assessment=" + assessment)
        result_.append(", Decision=" + decision)
        result_.append(", Issue=" + issue)
        result_.append("}")
        return result_.toString()
    }

    companion object {
        @JvmStatic
        fun toGenerateOutputData(other: Any?): GenerateOutputData? {
            if (other == null) {
                return null
            } else if (other is GenerateOutputData?) {
                return other
            } else if (other is com.gs.dmn.runtime.Context) {
                var result_ = GenerateOutputDataImpl()
                result_.decision = other.get("decision", "Decision") as String?
                result_.assessment = other.get("assessment", "Assessment") as java.math.BigDecimal?
                result_.issue = other.get("issue", "Issue") as java.math.BigDecimal?
                return result_
            } else if (other is com.gs.dmn.runtime.DMNType) {
                return toGenerateOutputData(other.toContext())
            } else if (other is proto.GenerateOutputData) {
                var result_: GenerateOutputDataImpl = GenerateOutputDataImpl()
                result_.decision = (other as proto.GenerateOutputData).getDecision()
                result_.assessment = java.math.BigDecimal.valueOf((other as proto.GenerateOutputData).getAssessment())
                result_.issue = java.math.BigDecimal.valueOf((other as proto.GenerateOutputData).getIssue())
                return result_
            } else {
                throw com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.javaClass.getSimpleName(), GenerateOutputData::class.java.getSimpleName()))
            }
        }

        @JvmStatic
        fun toProto(other: GenerateOutputData?): proto.GenerateOutputData {
            var result_: proto.GenerateOutputData.Builder = proto.GenerateOutputData.newBuilder()
            if (other != null) {
                var decisionProto_: String = (if ((other as GenerateOutputData).decision == null) "" else (other as GenerateOutputData).decision!!)
                result_.setDecision(decisionProto_)
                var assessmentProto_: Double = (if ((other as GenerateOutputData).assessment == null) 0.0 else (other as GenerateOutputData).assessment!!.toDouble())
                result_.setAssessment(assessmentProto_)
                var issueProto_: Double = (if ((other as GenerateOutputData).issue == null) 0.0 else (other as GenerateOutputData).issue!!.toDouble())
                result_.setIssue(issueProto_)
            }
            return result_.build()
        }

        @JvmStatic
        fun toProto(other: List<GenerateOutputData?>?): List<proto.GenerateOutputData>? {
            if (other == null) {
                return null
            } else {
                return other.stream().map({o -> toProto(o)}).collect(java.util.stream.Collectors.toList())
            }
        }
    }
}
